SimPEG/Directives.py

```diff
@@ -896,18 +881,30 @@ class UpdatePreconditioner(InversionDirective):
     """
     onlyOnStart = False
     mapping = None
-    ComboObjFun = False
+    misfitDiag = None
```

Comment:
would you mind adding docstrings to these `#: docstring here`. Also, would you mind updating the name to be more inline with pep8: `misfit_diag`